sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup  sys_xtimer Timers
+ * @ingroup   sys
+ * @brief     Provides a high level timer module to register
+ *            timers, get current system time, and let a thread sleep for
+ *            a certain amount of time.
+ *
+ * The implementation takes one low-level timer that is supposed to run at 1MHz
+ * speed and multiplexes it.
+ *
+ * Insertion and removal of timers has O(n) complexity with (n) being the
+ * number of active timers.  The reason for this is that multiplexing is
```

Comment:
> timer requirements are not crazy. Take a look at https://openwsn.atlassian.net/wiki/display/OW/Timing+Constants
timings of half a milisecond should well be within xtimer's
capabilities. Even 83us sounds like a doable precision even with more
active timers than the memory can fit.

>> > And, xtimer will not become deterministic just because the number of timers is limited.
> You stated something similar before, but still didn't explain it. What else is non-deterministic in the xtimers?

Well, this is probably a question of definition.

xtimer uses linked lists for timer management, and depending on the
number of timers in a given list, xtimer's cycle needs will vary. In my
definition, "determinstic" would mean, "xtimer always needs N cycles for
a given operation".

Limiting the number of active timers would give an upper bound of cycles
needed for some operations.